### ISSUE TITLE
coda_recognize_file_fuzzer: Integer-overflow in read_GDR

### ISSUE THREAD

**Issue by schwehr** [stcorp/coda#63]
```
third_party/stcorp_coda/libcoda/coda-cdf.c:1212:54: runtime error: signed integer overflow: 9223372036854775807 + 8 cannot be represented in type 'long'
    #0 0x5584b95f4871 in read_GDR third_party/stcorp_coda/libcoda/coda-cdf.c:1212:54
    #1 0x5584b95f3ae5 in read_file third_party/stcorp_coda/libcoda/coda-cdf.c:1442:9
    #2 0x5584b95f2be4 in coda_cdf_reopen third_party/stcorp_coda/libcoda/coda-cdf.c:1525:9
    #3 0x5584b95f2636 in reopen_with_backend third_party/stcorp_coda/libcoda/coda-product.c:398:17
    #4 0x5584b95ef328 in open_file third_party/stcorp_coda/libcoda/coda-product.c:552:9
    #5 0x5584b95eec6a in coda_recognize_file third_party/stcorp_coda/libcoda/coda-product.c:596:9
    #6 0x5584b9504af1 in LLVMFuzzerTestOneInput third_party/stcorp_coda/fuzz/coda_recognize_file_fuzzer.cc:19:3
```
The overflow happens here with `offset + 8`:
```c
static int read_GDR(coda_cdf_product *product_file, int64_t offset)
{
    int32_t record_type;
    int64_t rvdr_head;
    int64_t zvdr_head;
    int64_t adr_head;
    int64_t eof;
    int32_t nr_vars;
    int32_t num_attr;
    int32_t nz_vars;

    if (offset > read_bytes(product_file->raw_product, offset + 8, 4, &record_type) < 0)
    {
        return -1;
    }
```
Proposed fix.  This allows offsets that are insanely large, but prevents overflow.

```c
    if (offset > LONG_MAX - 8 ||
        read_bytes(product_file->raw_product, offset + 8, 4, &record_type) < 0)
    {
        return -1;
    }
```
[testcase-5313333665726464.zip](https://github.com/stcorp/coda/files/4581108/testcase-5313333665726464.zip)

**Comment by svniemeijer** [comment]
Should be solved with 9de7aa23435b702d771801090892e686053a6dbd

**Comment by schwehr** [comment]
Verified
